src/tests/ftest/util/general_utils.py

```diff
@@ -841,56 +841,53 @@ def dump_engines_stacks(hosts, verbose=True, timeout=60, added_filter=None):
-            values indicating which hosts yielded the return code.
-            Return code keys:
-                0   No processes matched the criteria / No processes killed.
-                1   One or more processes matched the criteria and a kill was
-                    attempted.
+        bool: True if processes matching the pattern where found on at least one host and a pkill
+            command was issued; False if no processes where found matching the pattern on any of the
+            hosts.
 
     """
-    result = {}
-    log = getLogger()
-    log.info("Killing any processes on %s that match: %s", hosts, pattern)
-
-    if added_filter:
-        ps_cmd = "/usr/bin/ps xa | grep -E {} | grep -vE {}".format(
-            pattern, added_filter)
+    kill_attempted = False
+    command = "/usr/bin/pgrep --list-full {}".format(pattern)
+    pattern_match = str(pattern)
+    if negative_filter:
+        command = "/usr/bin/ps xa | grep -E {} | grep -vE {}".format(pattern, negative_filter)
+        pattern_match += " and doesn't match " + str(negative_filter)
+
+    # Search for any active processes
+    log.debug("Searching for any processes on %s that match: %s", hosts, pattern_match)
+    result = run_remote(log, hosts, command, verbose, timeout)
+    if result.passed_hosts:
+        # Attempt to kill any processes found on any of the hosts with increasing force
+        steps = [("", 5), (" --signal ABRT", 1), (" --signal KILL", 0)]
+        while steps and result.passed_hosts:
+            step = steps.pop(0)
+            log.debug(
+                "Killing%s any processes on %s that match: %s",
+                step[0], result.passed_hosts, pattern_match)
+            kill_command = "sudo /usr/bin/pkill {}{}".format(step[0], pattern)
+            run_remote(log, result.passed_hosts, kill_command, verbose, timeout)
+            time.sleep(step[1])
+            result = run_remote(log, result.passed_hosts, command, verbose, timeout)
+        if result.passed_hosts:
+            log.warning(
+                "Processes still running on %s that match: %s", result.passed_hosts, pattern_match)
+        kill_attempted = True
     else:
-        ps_cmd = "/usr/bin/pgrep --list-full {}".format(pattern)
-
-    if hosts is not None:
-        commands = [
-            "rc=0",
-            "if " + ps_cmd,
-            "then rc=1",
-            "sudo /usr/bin/pkill {}".format(pattern),
-            "sleep 5",
-            "if " + ps_cmd,
-            "then sudo /usr/bin/pkill --signal ABRT {}".format(pattern),
-            "sleep 1",
-            "if " + ps_cmd,
-            "then sudo /usr/bin/pkill --signal KILL {}".format(pattern),
-            "fi",
-            "fi",
-            "fi",
-            "exit $rc",
-        ]
-        result = pcmd(hosts, "; ".join(commands), verbose, timeout, None)
-    return result
+        log.debug("No processes found on %s that match: %s", result.failed_hosts, pattern_match)
+    return kill_attempted
```

Comment:
I'd recommend moving this else above. Something like:
```
if not result.passed_hosts:
    log.debug("No processes found on %s that match: %s", result.failed_hosts, pattern_match)
    return False

# Now  don't need another indentation, or the `kill_attempted` variable

Reply:
Done and code has been moved to src/tests/ftest/util/run_utils.py